internal/webhook/apps/v1alpha1/nimservice_webhook_validation_helper.go

```diff
@@ -386,13 +393,45 @@ func validateScaleConfiguration(scale *appsv1alpha1.Autoscaling, fldPath *field.
 }
 
 // Spec.Resources.Claims must be empty.
-func validateResourcesConfiguration(resources *corev1.ResourceRequirements, fldPath *field.Path) field.ErrorList {
+func validateResourcesConfiguration(spec *appsv1alpha1.NIMServiceSpec, fldPath *field.Path) field.ErrorList {
 	errList := field.ErrorList{}
-	if resources != nil {
-		if resources.Claims != nil || len(resources.Claims) != 0 {
-			errList = append(errList, field.Forbidden(fldPath.Child("claims"), "must be empty"))
-		}
+
+	// Validate that Claims must be empty
+	if spec.Resources != nil && spec.Resources.Claims != nil && len(spec.Resources.Claims) > 0 {
```

Comment:
nit: if its nil, then len(...) will return 0
```suggestion
	if spec.Resources != nil && len(spec.Resources.Claims) > 0 {
```